acceptance/internal/resolve_server.go

```diff
@@ -0,0 +1,186 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+	"time"
+	"unicode/utf8"
+
+	"github.com/databricks/cli/libs/env"
+	"github.com/databricks/cli/libs/testserver"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func StartDefaultServer(t *testing.T) {
+	s := testserver.New(t)
+	addDefaultHandlers(s)
+
+	t.Setenv("DATABRICKS_DEFAULT_HOST", s.URL)
+
+	// Do not read user's ~/.databrickscfg
+	homeDir := t.TempDir()
+	t.Setenv(env.HomeEnvVar(), homeDir)
+}
+
+func isTruePtr(value *bool) bool {
+	return value != nil && *value
+}
+
+func ResolveServer(t *testing.T, config TestConfig, logRequests bool, outputDir string) *databricks.WorkspaceClient {
```

Comment:
I find name ResolveServer a bit odd. I understand StartServer is not descriptive because we're not always starting it, maybe PrepareServerAndClient?